proposals/threads/Overview.md

```diff
@@ -48,8 +48,48 @@ agent's cluster will have access to the additional linear memory.
 When a module has an imported linear memory, its data segments are copied into
 the linear memory when the module is instantiated.
 
-When the imported linear memory is shared, the writes are non-atomic and
-are not ordered.
+When linear memory is shared, it is possible for another module (or in the web 
+embedding, for JavaScript code) to read from the linear memory as it is being
+initialized.
+
+The data segments are initialized as follows, whether they apply to shared or
+non-shared linear memory:
+
+* Data segments are initialized in their definition order
+* From low to high bytes
+* At byte granularity (which can be coalesced)
+* As non-atomics
+* An entire module's data segment initialization then synchronizes with other
+  operations (effectively, followed by a barrier)
```

Comment:
I think you mean "and entire module's data *section* [...]" ? Since each data section contains multiple segments.